src/AiidaDFTK.jl

```diff
@@ -101,7 +102,36 @@ Run a DFTK calculation from a json input file.
 Output is by default written to `stdout` and `stderr`.
 The list of generated output files is returned.
 """
-function run_json(filename::AbstractString; extra_output_files=String[])
+function run_json(filename::AbstractString; extra_output_files=String[], min_version=typemin(VersionNumber), max_version=typemax(VersionNumber))
+    # We write a few key messages to an error file that we can parse from
+    # the AiiDA plugin to detect common failure modes.
+    errorfile = "errors.log"
+    if isfile(errorfile)
+        error("Error file $errorfile already exists!")
+    end
+    push!(extra_output_files, errorfile)
+    function report_error(msg)
+        mpi_master() || return
+        open(errorfile, "a") do io
+            println(io, msg)
+            flush(io)
```

Comment:
There should be no need. Once you go out of scope the file handle is flushed and closed.